### PR TITLE
Modify return null

### PR DESCRIPTION
**Pull request by lukyanoffpashok**

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#460](https://github.com/yegor256/aibolit/issues/460) is assigned to @aravij[/z](https://www.0crat.com/u/aravij), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/a0afc14a-c05a-4cdd-9931-38ec77a413cc); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: aibolit/patterns/return_null/return_null.py [hunk 1]
```diff
@@ -20,31 +20,41 @@
...
+from aibolit.utils.ast import AST, ASTNodeType
 
 
 class ReturnNull:
 
     def __init__(self):
         pass
 
-    def value(self, filename: str):
+    def get_return_args(self, node: int, tree: 'AST') -> bool:
```
Comment (aravij) [inline aibolit/patterns/return_null/return_null.py:33]: You shouldn't use quotes here `'AST'`. Quotes are used, when class you use is not defined yet.
This function name says it gets something, but it is actually check something. Please change the name.
You might make it private as well.

File: aibolit/patterns/return_null/return_null.py [hunk 2]
```diff
@@ -20,31 +20,41 @@
...
+        for child1 in tree.children_with_type(node, ASTNodeType.LITERAL):
+            for child2 in tree.children_with_type(child1, ASTNodeType.STRING):
+                if tree.get_attr(child2, 'string') == 'null':
+                    return True
+        return False
+
+    def work_with_return(self, node: int, lines: List[int], tree: 'AST') -> List[int]:
```
Comment (aravij) [inline aibolit/patterns/return_null/return_null.py:40]: Same, you shouldn't use quotes on types here and should make method private.
This method do two things at once: some AST checks and appends line number to list. This two functionality should be separated.
Method name also does not gives any hint to its task.

File: aibolit/patterns/return_null/return_null.py [hunk 3]
```diff
@@ -20,31 +20,41 @@
...
+        for child in tree.children_with_type(node, ASTNodeType.TERNARY_EXPRESSION):
+            if self.get_return_args(child, tree):
+                lines.append(tree.get_attr(node, 'source_code_line'))
+            return lines
```
Comment (aravij) [inline aibolit/patterns/return_null/return_null.py:44]: Seems like you leave method after you handle first child. This seems strange.

Comment (aravij) [reply aibolit/patterns/return_null/return_null.py:44]: Does the following code get executes, when there is no TERNARY_EXPRESSION children at all? If yes, we better make it more explicit, than now.

File: aibolit/patterns/return_null/return_null.py (hunk 1 above)
Comment (lukyanoffpashok) [reply aibolit/patterns/return_null/return_null.py:33]: fixed

File: aibolit/patterns/return_null/return_null.py (hunk 3 above)
Comment (lukyanoffpashok) [reply aibolit/patterns/return_null/return_null.py]: fixed

File: aibolit/patterns/return_null/return_null.py (hunk 2 above)
Comment (lukyanoffpashok) [reply aibolit/patterns/return_null/return_null.py]: fixed

**Comment by lukyanoffpashok** [comment]
@acheshkov please merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job audit: performer @aravij[/z](https://www.0crat.com/u/aravij)[/z](https://www.0crat.com/u/aravij) didn't make CR comments

**Pull request merged**